src/library/scala/collection/SeqLike.scala

```diff
@@ -232,7 +232,7 @@ trait SeqLike[+A, +Repr] extends Any with IterableLike[A, Repr] with GenSeqLike[
       if (idx < 0)
         _hasNext = false
       else {
-        var sum = nums.slice(idx + 1, nums.length).sum + 1
+        var sum = nums.view(idx + 1, nums.length).sum + 1
```

Comment:
I've pushed a new version with a plain-old loop. I've also added a unit test in a preceding commit to show the behaviour hasn't changed.